include/s3select.h

```diff
@@ -2420,6 +2492,13 @@ class base_s3object
         for (auto& a : *m_s3_select->get_aliases()->get())
         {
           a.second->invalidate_cache_result();
+          m_sa->set_column_name(a.first);
+          m_sa->set_alias(true);
+          col_ref = true;
+        }
```

Comment:
this loop should be executed only ones
the `*m_s3_select->get_aliases()->get()` never changed

`getMatchedRow` is an intensive function.